cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -188,7 +189,9 @@ protected void doShutdown() throws Exception {
   protected void runOneIteration() throws Exception {
     LOG.trace("Performing heartbeat publishing in Stream service instance {}", instanceId);
     ImmutableMap.Builder<String, Long> sizes = ImmutableMap.builder();
-    for (StreamSpecification streamSpec : streamMetaStore.listStreams()) {
+    Collection<StreamSpecification> specifications =
+      streamMetaStore.listStreams().get(new NamespaceMeta.Builder().setId(Constants.DEFAULT_NAMESPACE).build());
```

Comment:
Isn't it better you have a `listStreams` method in the meta store that supports listing per namespace?

```
listStreams()  // list all, regardless of namespace
listStreams(String namespace) // list all under the given namespace. The type may NamespaceMeta, which I am not sure
```